src/cloudai/configurator/gymnasium_adapter.py

```
@@ -0,0 +1,259 @@
+            else:
+                decoded[name] = self._decode_continuous(name, raw)
+        return decoded
+
+    def _decode_discrete(self, name: str, raw: Any) -> Any:
+        values = self._discrete_params[name]
+        idx = int(raw)
+        if not 0 <= idx < len(values):
+            raise ValueError(f"Action index out of range for '{name}': {idx} (expected 0..{len(values) - 1})")
+        return values[idx]
+
+    def _decode_continuous(self, name: str, raw: Any) -> Any:
```

Ruff (0.15.17)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `raw`

(ANN401)